cpp/src/ml_utils.h

```diff
@@ -28,7 +28,7 @@ namespace ML {
  * @param COV_EIG_JACOBI: covariance of input will be used along with eigen decomposition using jacobi method for symmetric matrices
  * @{
  */
-enum solver {
+enum class solver {
```

Comment:
While we're refactoring and moving these prims around, we should move this into include files and maybe create a base file for the shared parts. I don't think it needs to be done in this PR. Do you want to create an issue for this?